src/compiler/checker.ts

```diff
@@ -10663,7 +10730,17 @@ namespace ts {
                 });
             }
 
-            forEach(node.members, checkSourceElement);
+            forEach(node.members, node.flags & NodeFlags.Abstract ?
+                checkSourceElement :
+                element => {
+                    checkSourceElement(element);
+
+                    // Classes containing abstract methods must be marked abstract
+                    if (element.flags & NodeFlags.Abstract) {
+                        error(node, Diagnostics.Classes_containing_abstract_methods_must_be_marked_abstract);

```

Comment:
Okay I changed it to retain just that error then. It also is a bit more efficient (removes the extra iteration through the members).